tensorflow/contrib/learn/python/learn/learn_io/generator_io.py

```diff
@@ -0,0 +1,117 @@
+# Copyright 2016 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Methods to allow generator of dict with numpy arrays."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from types import FunctionType, GeneratorType
+from collections import OrderedDict
+
+from tensorflow.contrib.learn.python.learn.dataframe.queues import feeding_functions
+
+
+def generator_input_fn(x,
+                       target_key=None,
+                       batch_size=128,
+                       num_epochs=1,
+                       shuffle=True,
+                       queue_capacity=1000,
+                       num_threads=1):
+  """Returns input function that would feed a generator that yields dictionary
+  of numpy arrays into the model. It is assumed that every dict yielded from
+  the dictionary represents a single sample for every feature
+
+  This returns a function outputting `features` and `target` based on the dict
```

Comment:
I changed it so that the output is currently a feature dict, or  a feature dict and a target dict, if the target_key is used. I could cover the case where the target_key is a single str. In this case instead of a dict it returns a single tensor. This potentially makes it confusing as to what to expect from the function though as an output